test/e2e/basic_workflow_test.go

```diff
@@ -38,9 +38,14 @@ func TestBasicWorkflow(t *testing.T) {
 		test.serviceCreate(t, "hello")
 	})
 
+	t.Run("create hello service again and get service already exists error", func(t *testing.T) {
+		test.serviceCreateDuplicate(t, "hello")
```

Comment:
Nice nice, I wonder if we should also try to delete a service that’s already deleted? I would expect an error in case where service is already deleted (non-existent) and if the service is being deleted. Of course in the end the API should do the right thing and the client should return a “sensible” error.

Reply:
Yeah. Good point. I can add a test for deleting a non-existent service to this PR.